File: C++/Semana 3/Bubble.cpp
#include <iostream>
#include <vector>
using namespace std;

void bubbleSort(std::vector<int>& arr){
    int s = arr.size();
        for(int i = 0; i < s-1; i++){
            bool swapped = false;
            for(int j = 0; j < s-i-1; j++){
                if(arr[j] > arr[j+1]){
                    swap(arr[j], arr[j+1]);
                    swapped = true;
                }
            }
            if(!swapped){
                break;
            }
        }
        bool swapped = false;
}




int main() {
    std::vector<int> arr = {15,16,11,13,14};
    cout << "Antes de ordenar los elementos del array: " << endl;
    for(int i = 0; i < arr.size(); i++){
        cout << arr[i] << " ";
    }
    cout << endl;

    bubbleSort(arr);
    cout << "Despues de ordenar los elementos del array: " << endl;
    for(int i = 0; i < arr.size(); i++){
        cout << arr[i] << " ";
    }
    return 0;
}

File: Bubble.py
def bubblesort(a):
    s = len(a)
    for j in range(s):
        isSwapped = False
        for j in range(0, s - j - 1):
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]
                isSwapped = True
        if (isSwapped == False):
            break
if __name__ == "__main__":
    a = [15,16,11,13,14]
    print("Antes de ordenar los elementos del array: ")
    for j in a:
        print(j, end=' ')
    
    bubblesort(a)
    print("\nDespues de ordenar los elementos del array: " )
    for j in range(len(a)):
        print("%d" % a[j], end=" ")